src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * A storage class that contains methods for loading and saving tasks.
+ */
+
+package main.java.storage;
+
+import main.java.data.Deadline;
+import main.java.data.Event;
+import main.java.data.Task;
+import main.java.data.Todo;
+import main.java.exception.ChatbotException;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+public class Storage {
+
+    private final Path location;
+
+    public Storage(Path location) {
+        this.location = location;
+    }
+
+    /**
+     * Loads a list of tasks from a file on the disk. If no such file exists, create a new file.
+     * @return list of tasks saved to the disk
+     * @throws ChatbotException if could not load file
+     */
+    public ArrayList<Task> loadTasks() throws ChatbotException {
+        try {
+
+            // File does not exist
+            if (!Files.exists(location)) {
+                Files.createFile(location);
+                return  new ArrayList<>();
+            }
+
+            // File exists, load data
+            Stream<Task> taskStream = Files.lines(location).map(line -> {
+
+                String[] lineData = line.split("\\|");
+                String type = lineData[0].trim();
+                boolean isDone = lineData[1].trim().equals("1");
+                String description = lineData[2].trim();
+                String timestamp = lineData[3].trim();
+
+                Task task = null;
+
+                switch (type) {
+                    case "T":
+                        task = new Todo(description, isDone);
+                        break;
+                    case "D":
+                        task = new Deadline(description, isDone, LocalDate.parse(timestamp));
+                        break;
+                    case "E":
+                        task = new Event(description, isDone, LocalDate.parse(timestamp));
+                        break;
+                    default:
+                        break;
+                }
```

Comment:
I believe that the coding standard states that `case` clauses should not have indentation.